packages/yoastseo/src/languageProcessing/helpers/word/markWordsInSentences.js

```diff
@@ -19,10 +44,27 @@ export const collectMarkingsInSentence = function( sentence, topicFoundInSentenc
 	topicFoundInSentence = topicFoundInSentence.map( word => escapeRegExp( word ) );
 	// If a language has a custom helper to match words, we disable the word boundary when creating the regex.
 	const topicRegex = matchWordCustomHelper ? arrayToRegex( topicFoundInSentence, true ) : arrayToRegex( topicFoundInSentence );
-	const markup = sentence.replace( topicRegex, function( x ) {
+
+	// Retrieve the anchors and mark the anchors' text if the topic is found in the anchors' text.
+	const { anchors, markedAnchors } = getMarkedAnchors( sentence, topicRegex );
+
+	let markup = sentence.replace( topicRegex, function( x ) {
 		return addMark( x );
 	} );
 
+	/**
+	 * In 'markup', we apply the markings also inside the anchor's attribute if there is a match, on top of
+	 * marking the anchor's text.
+	 * The step below is to replace the incorrectly marked anchors with the marked anchors that we want:
+	 * where the markings are only applied in the anchor's text.
+	 */
+	if ( anchors.length > 0 ) {
+		const markupAnchors = getAnchorsFromText( markup );
+		for ( let i = 0; i < markupAnchors.length; i++ ) {
+			markup = markup.replace( markupAnchors[ i ], markedAnchors[ i ] );
+		}
+	}
```

Comment:
The approach is as follows:
* Get the anchors from the sentence
* For every anchor, apply the markings only to the anchor tag. Replace the unmarked anchor in the sentence with the marked anchor:
   * Retrieve the anchor text
   * Apply the marking to the anchor text
   * Replace the original anchor text with the marked anchor text
* If there is an anchor found in the sentence:
   * The incorrectly marked anchor will be replaced with the correctly marked anchor (only marked anchor text, excluding the attributes), retrieved from `getMarkedAnchors()`

This approach works, but I have to admit that it's a little bit convoluted.
Let me know if you have suggestions on how to improve the approach :)

Reply:
@agnieszkaszuba 's suggestion for improvement for this approach:
* Before running the functionality in `addMarkSingleWord`, get all the <a> tags (so only the opening tags) from the text and save them in an array (you can use this part of the regex from `getAnchorsFromText`: <a[\s]+(?:[^>]+)>)
* Apply marks to the text (`addMarkSingleWord`)
* Replace the <a> tags in the marked text with the original <a> tags

The advantage of this approach is that we don’t need to mark the text between alt tags twice, like it’s currently done in `addMarkSingleWord` and in `getMarkedAnchors`.

Reply:
We tried to implement the above suggestion. However, the third step "Replace the tags in the marked text with the original tags" proved to be tricky. And we couldn't formulate a regex that can detect anchor opening tag that includes the <yoastmark>.

Hence, we decided to use the current approach.